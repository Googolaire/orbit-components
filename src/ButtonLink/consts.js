// @flow
export const TYPES = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
};

export const TOKENS = {
  background: "background",
  backgroundHover: "backgroundHover",
  backgroundActive: "backgroundActive",
  backgroundFocus: "backgroundFocus",
  foreground: "foreground",
  foregroundHover: "foregroundHover",
  foregroundActive: "foregroundActive",
  foregroundFocus: "foregroundFocus",
  marginRightIcon: "marginRightIcon",
};
