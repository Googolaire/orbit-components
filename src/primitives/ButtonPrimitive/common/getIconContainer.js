// @flow
import { SIZE_OPTIONS, TOKENS } from "./consts";
import { rtlSpacing } from "../../../utils/rtl";
import { getSize } from "../../../Icon";

const getIconSpacing = ({ theme, right, size, onlyIcon }) => {
  if (onlyIcon) {
    return null;
  }
  const tokens = {
    [TOKENS.marginRightIcon]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.marginButtonIconLarge,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.marginButtonIconNormal,
      [SIZE_OPTIONS.SMALL]: theme.orbit.marginButtonIconSmall,
    },
  };
  if (right) {
    return rtlSpacing(`0 0 0 ${tokens[TOKENS.marginRightIcon][size]}`)({ theme });
  }
  return rtlSpacing(`0 ${tokens[TOKENS.marginRightIcon][size]} 0 0`)({ theme });
};

const getIconContainer = ({ onlyIcon, theme, right, size, sizeIcon }) => {
  return {
    spacing: getIconSpacing({ theme, right, size, onlyIcon }),
    foreground: null,
    foregroundHover: null,
    foregroundActive: null,
    height: getSize(sizeIcon)({ theme }),
    width: getSize(sizeIcon)({ theme }),
  };
};

export default getIconContainer;
