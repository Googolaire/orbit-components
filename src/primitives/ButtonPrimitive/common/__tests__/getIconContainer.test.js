// @flow
import getIconContainer from "../getIconContainer";
import theme from "../../../../defaultTheme";
import { SIZE_OPTIONS } from "../consts";
import { ICON_SIZES } from "../../../../Icon/consts";

describe("ButtonPrimitive: getIconContainer function", () => {
  it("should match when left icon", () => {
    const size = SIZE_OPTIONS.LARGE;
    const onlyIcon = false;
    const right = false;
    const sizeIcon = ICON_SIZES.MEDIUM;
    expect(getIconContainer({ onlyIcon, theme, right, size, sizeIcon })).toMatchSnapshot();
  });
  it("should match when right icon", () => {
    const size = SIZE_OPTIONS.SMALL;
    const onlyIcon = false;
    const right = true;
    const sizeIcon = ICON_SIZES.SMALL;
    expect(getIconContainer({ onlyIcon, theme, right, size, sizeIcon })).toMatchSnapshot();
  });
});
