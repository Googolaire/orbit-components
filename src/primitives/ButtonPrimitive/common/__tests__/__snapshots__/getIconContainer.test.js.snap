// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getIconContainer function should match when left icon 1`] = `
Object {
  "height": "24px",
  "margin": "0 12px 0 0",
  "width": "24px",
}
`;

exports[`ButtonPrimitive: getIconContainer function should match when right icon 1`] = `
Object {
  "height": "16px",
  "margin": "0 0 0 8px",
  "width": "16px",
}
`;
